hazelcast/src/test/java/com/hazelcast/test/OverridePropertyRule.java

```diff
@@ -0,0 +1,67 @@
+package com.hazelcast.test;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+
+/**
+ * Set or clear a property before running a test. The property will be restored once the test is finished.
+ *
+ */
+public final class OverridePropertyRule implements TestRule {
```

Comment:
What does this Rule do what our `Hazelcast*Runners` don't do? They already store and restore all system properties. So I think I'm missing something here :)

Reply:
I see the properties cleanup in runners as sort-of last resort. 

a test should not make assumptions about runners. it should not taint global environment and rely on someone else to clean it.

Reply:
I like the idea of a clean setup, but this is the only test using it, so it's more code style fragmentation. We should make a checklist for new tests and refactorings, with hints for the code parts, which are "silently" introduced over time:
* using mocked network? -> extend `HazelcastTestSupport` and its factory methods
* dealing with System properties? -> use `OverridePropertyRule`
* need to check for the root cause of a nested exception? -> use `RootCauseMatcher` with an `ExpectedException` rule
* is it a Near Cache test? -> use one of the unified abstract Near Cache test classes with a `NearCacheTestContext`
* ...